scripts/generate-custom-properties.js

```diff
@@ -109,12 +89,63 @@ Object.entries(definitions.plainTokens).forEach(([key, value]) => {
   tokens[key] = value;
 });
 
+// Write files
+const printJson = data =>
+  JSON.stringify(
+    Object.entries(data).reduce(
+      (acc, [key, value]) => ({
+        ...acc,
+        [`--${key}`]: value,
+      }),
+      {}
+    )
+  );
+
+const printJavaScript = data => `
+/*
+  THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+
+  This file is created by the 'scripts/generate-custom-properties.js' script.
+  The variables should be updated in 'materials/internals/definition.yaml'.
+*/
+
+${Object.entries(data)
+  .map(([key, value]) => `export const ${camelCase(key)} = "${value}";`)
+  .join('\n')}
+`;
+
+const printCss = data => `
+/*
+  THIS IS AN AUTOGENERATED FILE. DO NOT EDIT THIS FILE DIRECTLY.
+
+  This file is created by the 'scripts/generate-custom-properties.js' script.
+  The variables should be updated in 'materials/internals/definition.yaml'.
+*/
+
+:root {
+  ${Object.entries(data)
+    .map(([key, value]) => `--${key}: ${value};`)
+    .join('\n')}
+}
+`;
+
 fs.writeFileSync(
   path.join(__dirname, '../materials/custom-properties.json'),
-  JSON.stringify(prefixTokens(tokens), null, 2)
+  prettier.format(printJson(tokens), {
+    ...prettierConfig,
+    parser: 'json',
+  })
+);
+
+fs.writeFileSync(
+  path.join(__dirname, '../materials/custom-properties.js'),
+  prettier.format(printJavaScript(tokens), prettierConfig)
 );
 
 fs.writeFileSync(
   path.join(__dirname, '../materials/custom-properties.css'),
-  printCss(prefixTokens(tokens))
+  prettier.format(printCss(tokens), {
+    ...prettierConfig,
+    parser: 'css',
```

Comment:
This file is raw but 🤷‍♀️. Maybe another day.

Reply:
What do you mean?

Reply:
Nevermind, non clearly communicated feeling of mine that this file could use some attention and care at some point.